doc/nat/nat_design_spec.md

```diff
@@ -0,0 +1,1550 @@
+
+NAT translation of the ICMP traffic is always done in the Linux kernel. ICMP traffic crossing the zones is trapped to the CPU. ICMP NAT/NAPT entries hence are not added to hardware.
+
+Using tools like Ping and Traceroute from internal hosts destined to external public IP addresses should work with NAT translations.
+
+The ICMP error messages received from the external public IP networks in response to the packets from internal hosts, are NAT/NAPT translated back to the internal hosts.
+The NAT translation for the ICMP error messages is based on RFC 5508, Section 4.2, where the contents of the embedded ICMP payload's IP header and transport headers are inspected to do DNAT translation of the outer IP header and the embedded IP and transport headers.
+
+The ICMP NAT session timeout is 30 seconds as maintained by the Kernel.
+
+## 3.5 Docker for NAT
+NatSyncd and NatMgrd daemons run in a separate docker named 'nat'. It can be stopped/started/restarted independently.
+
+```
+root@sonic:/home/admin# docker ps
+CONTAINER ID        IMAGE                             COMMAND                  CREATED              STATUS              PORTS               NAMES
+9ff843cc79fa        docker-syncd-brcm:latest          "/usr/bin/supervisord"   About a minute ago   Up About a minute                       syncd
+042faedb7b4b        docker-dhcp-relay:latest          "/usr/bin/docker_iniâ¦"   About a minute ago   Up About a minute                       dhcp_relay
+210f0ad0b776        docker-router-advertiser:latest   "/usr/bin/supervisord"   About a minute ago   Up About a minute                       radv
+b45498e6e705        docker-orchagent-brcm:latest      "/usr/bin/supervisord"   About a minute ago   Up About a minute                       swss
+a77dbe67649a        docker-nat:latest                 "/usr/bin/supervisord"   2 minutes ago        Up About a minute                       nat
+0d3b86e300f9        docker-lldp-sv2:latest            "/usr/bin/supervisord"   2 minutes ago        Up About a minute                       lldp
+f65865e90fd1        docker-platform-monitor:latest    "/usr/bin/supervisord"   2 minutes ago        Up About a minute                       pmon
+4374ebb6e0e1        docker-teamd:latest               "/usr/bin/supervisord"   2 minutes ago        Up About a minute                       teamd
+6244613730fb        docker-fpm-frr:latest             "/bin/sh -c '/usr/biâ¦"   2 minutes ago        Up About a minute                       bgp
+9ba9d3e63426        docker-database:latest            "/usr/bin/supervisord"   2 minutes ago        Up 2 minutes                            database
+```
+## 3.6 SAI
+Table shown below represents the SAI attributes which shall be used for NAT.
+
+NOTE: NAT SAI Spec is not finalized at the time this spec is written, and is subject to change. The PR for the NAT SAI Spec is @ [NAT SAI Spec PR](https://github.com/opencomputeproject/SAI/pull/937)
```

Comment:
Seems that NAT SAI spec PR is merged, could you please update the design accordingly?

Reply:
Sure.

Reply:
@kirankella, wondering if you get a chance to verify/update the hld any change here with latest SAI spec?  Thanks.